galsim/bandpass.py

```diff
@@ -463,12 +430,49 @@ def thin(self, rel_err=1.e-4, preserve_range=False):
             f = self(x)
             newx, newf = utilities.thin_tabulated_values(x, f, rel_err=rel_err,
                                                          preserve_range=preserve_range)
-            # preserve type
-            ret = self.copy()
-            ret.func = galsim.LookupTable(newx, newf, interpolant='linear')
-            ret.blue_limit = np.min(newx)
-            ret.red_limit = np.max(newx)
-            ret.wave_list = np.array(newx)
-            if hasattr(ret, '_effective_wavelength'):
-                del ret._effective_wavelength
-            return ret
+            tp = galsim.LookupTable(newx, newf, interpolant='linear')
+            blue_limit = np.min(newx)
+            red_limit = np.max(newx)
+            wave_list = np.array(newx)
+            return Bandpass(tp, blue_limit, red_limit, _wave_list=wave_list)
+        else:
+            return self
+
+    def __eq__(self, other):
+        return (isinstance(other, Bandpass) and
+                self._orig_tp == other._orig_tp and
+                self.red_limit == other.red_limit and
+                self.blue_limit == other.blue_limit and
+                self.wave_factor == other.wave_factor and
+                self.zeropoint == other.zeropoint and
+                (self.wave_list == other.wave_list).all())
+
+    def __getstate__(self):
+        d = self.__dict__.copy()
+        del d['_tp']
```

Comment:
By deleting `_tp` and `func`, I guess it's not possible for someone to pickle a `Bandpass` (or `SED`) initialized from an ascii file, send it to a friend, and then have them be able to unpickle it (unless they have the same ascii file at the same path).  Maybe this is worth mentioning somewhere?

Alternatively, is there are reason not to pickle the LookupTable if that's how the throughput/flambda is being represented (other than being unwieldy)?